### ISSUE TITLE
Remote ollama host - ssl error, authentication?

### ISSUE THREAD

**Issue by spaasis** [paul-gauthier/aider#1123]
### Issue

Hi! Aider seems promising, but so far I haven't managed to get it up and running.

TL;DR:
- `--no-verify-ssl` does not seem to work for ollama
- how to set Authorization header for ollama requests?

We're running OpenWebUI ( https://docs.openwebui.com/) and it's providing an ollama compatible API over at https://myaiservice.domain/ollama.

I've configured OLLAMA_API_BASE env variable to point to the above address, and am running `aider --model ollama/llama3.1:latest --no-verify-ssl`, Despite the flag to skip ssl verification, any request ends in `Error during API call: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate
(_ssl.c:1000)`

(for reference - the same flag seems to work fine when calling openai compatible APIs)

But even if I got the connection working, how would I configure the authentication headers? The API calls require `Authorization` header as `Bearer eyj...`. Browsing the docs I couldn't find anything about customizing the headers of the API calls. For OpenAI APIs this can be achieved via `OPENAI_API_KEY` but is there a similar option for Ollama?

### Version and model info

Aider 0.50.1 on Windows

**Comment by paul-gauthier** [comment]
Thanks for trying aider and filing this issue.

Aider supports ollama via litellm. It looks like `OLLAMA_API_KEY` may be a way to set the key.

You may want to open an issue with them about these detailed questions?

https://docs.litellm.ai/docs/providers/ollama

**Comment by spaasis** [comment]
@paul-gauthier thanks for the quick reply. Seeing as this is at most a documentation issue at this repo, I'll close this issue and browse the docs and issues for litellm

**Comment by dannylank** [comment]
It doesn't work with anthropic either:
`
aider --git --browser --no-auto-commits --model anthropic/claude-3-5-sonnet-20240620  --no-verify-ssl

Watching for file changes.

  You can now view your Streamlit app in your browser.

  Local URL: http://localhost:8501
  Network URL: http://192.168.1.14:8501

Aider v0.51.2-dev
Main model: anthropic/claude-3-5-sonnet-20240620 with diff edit format, infinite output
Weak model: claude-3-haiku-20240307
Git repo: ..\..\.git with 887 files
Repo-map: using 1024 tokens, auto refresh

litellm.InternalServerError: AnthropicException - [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate in certificate chain (_ssl.c:1000). Handle with `litellm.InternalServerError`.
Retrying in 0.2 seconds...
litellm.InternalServerError: AnthropicException - [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate in certificate chain (_ssl.c:1000). Handle with `litellm.InternalServerError`.
Retrying in 0.5 seconds...
litellm.InternalServerError: AnthropicException - [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate in certificate chain (_ssl.c:1000). Handle with `litellm.InternalServerError`.
Retrying in 1.0 seconds...
litellm.InternalServerError: AnthropicException - [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate in certificate chain (_ssl.c:1000). Handle with `litellm.InternalServerError`.
Retrying in 2.0 seconds...
litellm.InternalServerError: AnthropicException - [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self-signed certificate in certificate chain (_ssl.c:1000). Handle with `litellm.InternalServerError`.`
